tap_mongodb/connector.py

```diff
@@ -116,7 +118,14 @@ def discover_catalog_entries(self) -> List[Dict[str, Any]]:
             The discovered catalog entries as a list.
         """
         result: List[Dict] = []
-        for collection in self.database.list_collection_names(authorizedCollections=True, nameOnly=True):
+
+        collections = self.database.list_collection_names(
+            authorizedCollections=True,
+            nameOnly=True,
+            filter={"$or": [{"name": c} for c in self._collections]} if self._collections else None,
```

Comment:
```suggestion
            filter={"name": {"$in": self._collections}} if self._collections else None,
```

This would be ideal, but I get the following error:

```
pymongo.errors.OperationFailure: can't get regex from filter doc not a regex, full error: {'ok': 0, 'errmsg': "can't get regex from filter doc not a regex", 'code': 8000, 'codeName': 'AtlasError'}
```

Can't see any indication from docs that this isn't supported:
- https://www.mongodb.com/docs/manual/reference/command/listCollections/
- https://www.mongodb.com/docs/compass/current/query/filter/
- https://www.mongodb.com/docs/manual/reference/operator/query/in/